python/pyspark/sql/functions.py

```diff
@@ -101,19 +101,19 @@ def lit(col):
 
 
 @since(1.3)
-def col(col):
+def col(col_like):
```

Comment:
```suggestion
def col(col):
```

Reply:
Don't you think `col_like` is a better name? It could be column name as a string or a Column object - I see a comment at the beginning of the file. Besides, it will avoid another ambiguous event like this.